### ISSUE TITLE
Patch to fix code warnings

### ISSUE THREAD

**Issue by pronovic** [vapier/ncompress#2]
Attached is code-warnings.patch from Debian package 4.2.4.4-10.  It fixes a code warning in compress42.c.

KEN

```
# Description: Fix compiler warnings in the code.
# Author: Kenneth J. Pronovici <pronovic@debian.org>
Index: ncompress-4.2.4.4/compress42.c
===================================================================
--- ncompress-4.2.4.4.orig/compress42.c
+++ ncompress-4.2.4.4/compress42.c
@@ -1007,8 +1007,12 @@ comprexx(fileptr)

               if (infstat.st_nlink > 1 && (!force))
               {
-                 fprintf(stderr, "%s has %d other links: unchanged\n",
-                             tempname, infstat.st_nlink - 1);
+                 /* There's no portable way to print an nlink_t, because it
+                     changes size by platform.  So, I just cast to unsigned
+                     long int and hope for the best.  This is basically the
+                     StackOverflow recommendation: http://stackoverflow.com/questions/1401526 */
+                 fprintf(stderr, "%s has %ld other links: unchanged\n",
+                             tempname, (unsigned long int) infstat.st_nlink - 1);
                  exit_code = 1;
                  return;
               }
```

**Comment by pronovic** [comment]
This patch is included in the pull request referenced above.

**Comment by vapier** [comment]
as mentioned in the pull request, using long for nlink_t won't work as some platforms have 8 byte nlink_t and 4 byte long (like the x32 ABI)

fixed w/intmax_t in commit add92bc8f2061700872f770ef99c7b9c8beb028d

**Comment by pronovic** [comment]
Ok, make sense.  I had tested i386 and amd64.  I'll adjust the Debian patch to match your implementation.
